workflow-runtime/src/commonMain/kotlin/com/squareup/workflow1/WorkflowInterceptor.kt

```diff
@@ -129,6 +142,21 @@ public interface WorkflowInterceptor {
     session: WorkflowSession
   ): Snapshot? = proceed(state)
 
+  /**
+   * Called when a [RenderingAndSnapshot] is updated from the runtime (i.e. when the [StateFlow]
+   * is provided a new value - this may still be de-duplicated).
+   *
+   * @return [RenderingAndSnapshot]: a possibly modified [RenderingAndSnapshot].
+   */
+  public fun <R> onRenderingUpdated(
+    rendering: RenderingAndSnapshot<R>
+  ): RenderingAndSnapshot<R> = rendering
+
+  /**
+   * Called when the rendering is skipped because no state has changed after action cascade(s).
+   */
+  public fun onRenderingSkipped(): Unit = Unit
```

Comment:
if the outcome of an action loop is always either "onRenderingSkipped" or "onRenderingUpdated", then I would actually merge those and pass in a sealed class of "Updated" vs "Skipped", which would make this behavior a lot clearer.

Reply:
ya I think that sounds reasonable!